### PR TITLE
Fixes 3600: delete distributions and env on template delete

### PR DESCRIPTION
**Pull request by rverdile**
## Checklist

- [ ] Tested with snapshotting feature disabled and pulp server URL not configured if appropriate

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-3600

File: pkg/tasks/delete_repository_snapshots.go [hunk 1]
```diff
@@ -172,3 +185,42 @@ func (d *DeleteRepositorySnapshots) deleteRepoConfig() error {
 	}
 	return nil
 }
+
+func (d *DeleteRepositorySnapshots) deleteCandlepinContent() error {
```
Comment (jlsherrill) [inline pkg/tasks/delete_repository_snapshots.go:189]: i'd return early if candlepin isn't configured

**Comment by swadeley** [comment]
Hi

Please rebase to get fix in https://github.com/content-services/content-sources-backend/pull/697

**Review by jlsherrill — APPROVED** [review]

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-4163

**Pull request merged**